project.md

```diff
@@ -11,41 +11,79 @@ _What need is met by your idea? Who are you helping? What is that person's
+	This led me to the idea of making a DSL by limiting the features a lot of calendar apps have, and by doing so creating a robust DSL that could quickly create new schedules and events. I’m ultimately excited about this idea because I personally would use it. I think it would help me be more organized and manage my time better, and if there are people who feel similarly with their calendar app, it would be amazing if this could help them make schedules as well.
+
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Making schedules easier and faster
+
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+While I haven’t finalized the syntax yet, I hope the user will interact with the language by writing out a text file. This could be done in any text editor, or in a command line terminal, or in some type of preview option. The user would write out a line for a single event, specify some arguments like start/end time, start/end day, title, description, and what days it repeats on. I also hope to incorporate natural language, so a user could for instance type out “set ‘CS111’ describing ‘Domain Specific Languages in Shanahan 2465’ on Monday Wednesday from 1:15pm to 2:30pm”. But I also hope to give the option for the user to not need natural language, typing for that same event “‘CS111’ ‘Domain Specific Languages in Shanahan 2465’ M W 1:15pm 2:30pm”. I think this is the right way to interact with the problem domain because this will create an event much quicker than interacting with a calendar app. I also think it’s much more intuitive than Icalendar.
+
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+While I haven’t finalized the syntax/semantics of the DSL yet, I would like to use an external DSL using parser combinators to read in the user input. The program the user writes would then be read, and then this could either result in an Icalendar ICS file, or it could be passed to a second DSL which will visualize it in a HTML/Excel file. 
+	Some errors that might occur could be type errors, such as inputting “29:72pm” for the start time, or syntax errors such as not giving a title for the event. I think this could be avoided by simply catching and returning errors for these. These could then be communicated to the user via error messages. I know there are probably more errors that are possible, but I can not think of any because it is 3:15 am and I am very tired. 
+
```

Comment:
My biggest question is How the calendar/scheduler gets visualized. Exporting it to another calendar type makes sense so that you can see the calendar easy without having to go through all the complicated stuff to make an event. But it also makes sense to have your own visualization - I'm curious what that end-visualization would look like. Like a weekly calendar? A database (the excel file?) or more just like a list of events.